pinot-core/src/main/java/org/apache/pinot/core/query/postaggregation/PostAggregationFunction.java

```diff
@@ -49,24 +51,27 @@ public PostAggregationFunction(String functionName, ColumnDataType[] argumentTyp
       }
     }
     _functionInvoker = new FunctionInvoker(functionInfo);
-    Class<?>[] parameterClasses = _functionInvoker.getParameterClasses();
-    PinotDataType[] parameterTypes = _functionInvoker.getParameterTypes();
-    int numArguments = argumentTypes.length;
-    int numParameters = parameterClasses.length;
-    Preconditions.checkArgument(numArguments == numParameters,
-        "Wrong number of arguments for method: %s, expected: %s, actual: %s", functionInfo.getMethod(), numParameters,
-        numArguments);
-    for (int i = 0; i < numParameters; i++) {
-      Preconditions.checkArgument(parameterTypes[i] != null, "Unsupported parameter class: %s for method: %s",
-          parameterClasses[i], functionInfo.getMethod());
-    }
-    _argumentTypes = new PinotDataType[numArguments];
-    for (int i = 0; i < numArguments; i++) {
-      _argumentTypes[i] = PinotDataType.getPinotDataTypeForExecution(argumentTypes[i]);
-    }
     ColumnDataType resultType = FunctionUtils.getColumnDataType(_functionInvoker.getResultClass());
     // Handle unrecognized result class with STRING
     _resultType = resultType != null ? resultType : ColumnDataType.STRING;
+
+    if (!_functionInvoker.getMethod().isVarArgs()) {
```

Comment:
I understand this check, but is there a reason why we need to move the code after `resultType` is declared? As far as I can see the rest of the code is the same, right?

Reply:
I think the diff is slightly confusing, all I did was to separate the common parts and the part which should only be executed when the scalar function is not varargs.